frontend/src/pages/StudentProfile.vue

```diff
@@ -70,18 +71,17 @@ export default {
         },
       },
       studentHistory: [],
+       studentID: this.$route.params.id,
     };
   },
   mounted: function() {
     const id = this.$route.params.id
-    fetch(`/api/students/${id}`)
-      .then(response => response.json())
-      .then(data => {
-        const studentObject = {
+    const data = this.$store.getters.getStudentByStudentId(id)
```

Comment:
can we use vuex helpers? for the getters just use `mapGetters` then use it as computed property
```javascript
import { mapGetters  } from 'vuex'

{
  computed: {
   ...mapGetters([ 'getStudentByStudentId' ])
  }
}
```

how to use
```javascript
mounted: {
  const data = this.getStudentByStudentId(id)
}
```

Reply:
Thank you for the guide on mapGetters :) Done!